crates/openvino/src/element_type.rs

```diff
@@ -1,77 +1,59 @@
-    I16 = 9,
+    I16 = openvino_sys::ov_element_type_e_I16,
     /// A 32-bit integer element type.
-    I32 = 10,
+    I32 = openvino_sys::ov_element_type_e_I32,
     /// A 64-bit integer element type.
-    I64 = 11,
+    I64 = openvino_sys::ov_element_type_e_I64,
     /// An 1-bit unsigned integer element type.
-    U1 = 12,
+    U1 = openvino_sys::ov_element_type_e_U1,
     /// An 4-bit unsigned integer element type.
-    U4 = 13,
+    U4 = openvino_sys::ov_element_type_e_U4,
     /// An 8-bit unsigned integer element type.
-    U8 = 14,
+    U8 = openvino_sys::ov_element_type_e_U8,
     /// A 16-bit unsigned integer element type.
-    U16 = 15,
+    U16 = openvino_sys::ov_element_type_e_U16,
     /// A 32-bit unsigned integer element type.
-    U32 = 16,
+    U32 = openvino_sys::ov_element_type_e_U32,
     /// A 64-bit unsigned integer element type.
-    U64 = 17,
+    U64 = openvino_sys::ov_element_type_e_U64,
     /// NF4 element type.
-    NF4 = 18,
+    NF4 = openvino_sys::ov_element_type_e_NF4,
     /// F8E4M3 element type.
-    F8E4M3 = 19,
+    F8E4M3 = openvino_sys::ov_element_type_e_F8E4M3,
     /// F8E5M3 element type.
-    F8E5M3 = 20,
+    F8E5M3 = openvino_sys::ov_element_type_e_F8E5M3,
 }
 
-#[cfg(test)]
-mod tests {
-    use super::*;
-    use openvino_sys::*;
+const LARGEST_VARIANT: u32 = openvino_sys::ov_element_type_e_F8E5M3;
 
-    #[test]
-    fn check_discriminant_values() {
-        assert_eq!(ov_element_type_e_UNDEFINED, ElementType::Undefined as u32);
-        assert_eq!(ov_element_type_e_U1, ElementType::U1 as u32);
-        assert_eq!(ov_element_type_e_U4, ElementType::U4 as u32);
-        assert_eq!(ov_element_type_e_U8, ElementType::U8 as u32);
-        assert_eq!(ov_element_type_e_U16, ElementType::U16 as u32);
-        assert_eq!(ov_element_type_e_U32, ElementType::U32 as u32);
-        assert_eq!(ov_element_type_e_U64, ElementType::U64 as u32);
-        assert_eq!(ov_element_type_e_I4, ElementType::I4 as u32);
-        assert_eq!(ov_element_type_e_I8, ElementType::I8 as u32);
-        assert_eq!(ov_element_type_e_I16, ElementType::I16 as u32);
-        assert_eq!(ov_element_type_e_I32, ElementType::I32 as u32);
-        assert_eq!(ov_element_type_e_I64, ElementType::I64 as u32);
-        assert_eq!(ov_element_type_e_F16, ElementType::F16 as u32);
-        assert_eq!(ov_element_type_e_F32, ElementType::F32 as u32);
-        assert_eq!(ov_element_type_e_F64, ElementType::F64 as u32);
-        assert_eq!(ov_element_type_e_OV_BOOLEAN, ElementType::Boolean as u32);
-        assert_eq!(ov_element_type_e_DYNAMIC, ElementType::Dynamic as u32);
-        assert_eq!(ov_element_type_e_NF4, ElementType::NF4 as u32);
-        assert_eq!(ov_element_type_e_F8E4M3, ElementType::F8E4M3 as u32);
-        assert_eq!(ov_element_type_e_F8E5M3, ElementType::F8E5M3 as u32);
+impl From<u32> for ElementType {
+    fn from(value: u32) -> Self {
+        if value > LARGEST_VARIANT {
+            return ElementType::Undefined;
```

Comment:
This might be better as a `TryFrom` implementation: if the user passes in a `u32` value that doesn't line up, we can then return an error instead of silently converting it into `ElementType::Undefined`.

Reply:
Alternately, do we _need_ this "from `u32`" functionality?

Reply:
I think `From` was needed before [this commit](https://github.com/intel/openvino-rs/pull/99/commits/92c9e057d294161ee35e5244cb9055e8f35750d2). The `into()` made sense when [this](https://github.com/intel/openvino-rs/blob/92c9e057d294161ee35e5244cb9055e8f35750d2/crates/openvino/src/tensor.rs#L88) was `let mut element_type = 0`, but I have it initialized as `ElementType::Undefined`.
So probably can do without `From` implemented.